code/game/gamemodes/miniantags/revenant/revenant.dm

```diff
@@ -121,9 +121,20 @@
 	remove_from_all_data_huds()
 
 	spawn(5)
-		if(src.mind)
-			src.mind.wipe_memory()
```

Comment:
They'll retain previous goals if they were already alive in the round no? Why not wipe the memory?